invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/frontpage/records-list.js

```diff
@@ -0,0 +1,90 @@
+// This file is part of InvenioRDM
+// Copyright (C) 2022 CERN.
+//
+// Invenio RDM is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { i18next } from "@translations/invenio_app_rdm/i18next";
+import { withCancel } from "react-invenio-forms";
+import { Loader, Container, Header, Item, Button } from "semantic-ui-react";
+import { RDMRecordResultsListItem } from "../search/components";
+import axios from "axios";
+
+export default class RecordsList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: { hits: [] },
+      isLoading: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentWillUnmount() {
+    this.cancellableFetch && this.cancellableFetch.cancel();
+  }
+
+  fetchData = async () => {
+    this.setState({ isLoading: true });
+
+    this.cancellableFetch = withCancel(
+      axios.get(this.props.fetchUrl, {
+        headers: {
+          Accept: "application/vnd.inveniordm.v1+json",
+        },
+        withCredentials: true,
+      })
+    );
+
+    try {
+      const response = await this.cancellableFetch.promise;
+      this.setState({ data: response.data.hits, isLoading: false });
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  renderItems() {
```

Comment:
this part does not need to be moved to a separate method, can be done inside the render function